pysrc/hdmi_switcher_server.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+import os
+import sys
+import json
+import serial
+from flask import Flask, request
+import logging
+from logging.config import dictConfig
+
+config_file = os.path.join(os.path.dirname(__file__),
+                '../config/serial-settings.json')
+
+if not os.path.isfile(config_file):
+    raise RuntimeError("Config file {0} not found".format(config_file))
+
+with open(config_file) as f:
+    config = json.load(f)
+    print('Loaded config file {0}'.format(config_file))
+
+dictConfig({
+    'version': 1,
+    'formatters': {'default': {
+        'format': '[%(asctime)s] %(levelname)-7s %(module)-22s: %(message)s',
+    }},
+    'handlers': {'wsgi': {
+        'class': 'logging.StreamHandler',
+        'stream': 'ext://flask.logging.wsgi_errors_stream',
+        'formatter': 'default'
+    }},
+    'root': {
+        'level': config['server']['logging_level'],
+        'handlers': ['wsgi']
+    }
+})
+
+app = Flask(__name__)
+
+
+@app.route('/', methods=['POST'])
+def handle_serial():
+    request_json = request.get_json()
+    app.logger.debug('Request JSON {0}'.format(request_json))
+
+    with open(config_file) as f:
+        config = json.load(f)
+    
+    log_level = 'ERROR'
+    if u'device' not in request_json.keys():
+        response = "FAILURE. Key 'device' not in JSON payload {0}".format(
+            request_json)
+    else:
+        selected_device_name = request_json[u'device'].lower()
+
+        if selected_device_name not in config['hdmi_devices_map']:
+            response = "FAILURE. Requested device '{0}' not defined. Please " \
+                "select from '{1}'".format(selected_device_name, 
+                "', '".join(list(config['hdmi_devices_map'].keys())))
```

Comment:
Does the string `join` fail without the list? I thought the `dict.keys()` returned an iterable of strings? Could be wrong 🤷🏾

Reply:
It does not fail when `dict.keys()` is empty. `','.join({}.keys())` -> `''`